pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -1205,19 +1205,24 @@ public boolean tryLoadExistingSegment(SegmentZKMetadata zkMetadata, IndexLoading
        - The "DONE" status confirms the COMMIT_END_METADATA call succeeded,
          and the segment is available either in deep storage or with a peer
          before discarding the local copy.
+       The only exception is if the server does not check CRC on segment load.
 
     Then:
     We need to fall back to downloading the segment from deep storage to load it.
     */
     if (segmentMetadata == null || (isSegmentStatusCompleted(zkMetadata) && !hasSameCRC(zkMetadata, segmentMetadata))) {
       if (segmentMetadata == null) {
         _logger.info("Segment: {} does not exist", segmentName);
-      } else if (!hasSameCRC(zkMetadata, segmentMetadata)) {
-        _logger.info("Segment: {} has CRC changed from: {} to: {}", segmentName, segmentMetadata.getCrc(),
-            zkMetadata.getCrc());
+        closeSegmentDirectoryQuietly(segmentDirectory);
```

Comment:
nit: we can also cleanup the code a bit and hoist the `segmentMetadata == null` check.

Reply:
Good point. Revised.